src/Misc/WaveShapeSmps.cpp

```diff
@@ -13,6 +13,7 @@
 
 #include "WaveShapeSmps.h"
 #include <cmath>
+#define X smps[i]
```

Comment:
Yeah, I'm going to recommend against this. If you have loops where the expression is used a number of times feel free to have "float &X = smps[i];", but listing this as a macro across the whole file makes it easy to mistakenly use and makes some of the code marginally less obvious.

That said, I do like that it makes some the further downwind changes easier to read, so as long as there's the "float &X = smps[i]" at the right scope, the smps[i]->X changes seem to be a net gain in readability.